### ISSUE TITLE
Data collection for fine-tuning specific tasks.

### ISSUE THREAD

**Issue by SilentJCR** [NVIDIA/Isaac-GR00T#715]
I am collecting data of picking up cubes in various colors (e.g., red, green, and blue) to train a GR00T-based model for my robotic arm. In the current dataset, each episode contains at least one target cube that the arm needs to pick up.

After fine-tuning the model for 5,000 to 10,000 steps, I noticed that during inference, the arm sometimes picks up a cube of the wrong color.

To address this misclassification/wrong-picking issue, I am wondering what the root cause might be. I am also considering introducing negative episodes into the training data. For example, if the task command is "pick up the red cube" but no red cube is present on the table, the arm should remain stationary (execute no-op/idle actions) instead of moving.

Would this approach effectively mitigate the wrong-picking behavior, or are there better ways to handle color-conditioning in GR00T?

**Comment by praveenkathirvel-etr** [comment]
Currently, I am using negative sampling to handle this issue. In your context that would be to add episodes where there is no red cube and the bot actually does nothing, this has given me varying level of success. But I do not know how you can inherently add the thinking logic without just using a hack to bypass semantic understanding.

**Comment by SilentJCR** [comment]
> Currently, I am using negative sampling to handle this issue. In your context that would be to add episodes where there is no red cube and the bot actually does nothing, this has given me varying level of success. But I do not know how you can inherently add the thinking logic without just using a hack to bypass semantic understanding.

Varying levels of success? Does that mean it actually still helped? We came up with this idea because in the past when training object detection models, users could specify regions and label them as the background class so that object detection models are less likely to falsely detect regions that look like what they are supposed to detect.

Aside from avoiding picking up cubes of the wrong colors, we also hope that after picking up all the cubes of the specific color in sight, the arm should just return to its starting position and stay stationary instead of moving back and forth as if it still intended to pick something up and even doing so as a result.

**Comment by praveenkathirvel-etr** [comment]
It did actually help. The behavior and the approach you mentioned for training object detection models is pretty much exactly our idea of doing this. So in a simpler context. In a sample jug pouring task; w/o Negatives the robot would just try and grasp something where it thinks the jug should be or moves toward that direction and model collapses. w/Negatives in the absence of the object of interest it would just stay in the starting position i.e, do nothing. 

One keen observation is to note is the state_dropout parameter. I am pretty sure nVidia intended for the end user to collect as little data as possible and still run the VLA with success. So nVidia has set it to 0.2 by default. But if you are to collect 100s and 1000s of episodes I'd recommend using a lower number so that it won't over-fit to actions that are in our data_collection. Trick is to find an ideal number, as this directly affects how much pro-prioceptive data we want to fine-tune on. At a very low number this learns a general action and keeps repeating it regardless of what the world/image looks like. So to achieve a balance in b/w both visual and pro-prioceptive data; For this you need to set the bounds of your task properly; it could be a stable environment with little to no dynamics or very dynamic env with active human/external perturbation in which case you would need the model to be more biased on the visual data (higher state_dropout).
